tests/test_piped.py

```diff
@@ -86,11 +88,24 @@ def available_bzip2_readers_and_writers():
 
 PIPED_BZIP2_READERS, PIPED_BZIP2_WRITERS = available_bzip2_readers_and_writers()
 
-ALL_READERS_WITH_EXTENSION = list(zip(PIPED_GZIP_READERS, cycle([".gz"]))) + list(
-    zip(PIPED_BZIP2_READERS, cycle([".bz2"]))
+
+def available_xz_readers_and_writers():
+    if shutil.which("xz"):
+        return [PipedXzReader], [PipedXzWriter]
+    return [], []
```

Comment:
The coverage decreases because this line is never called. It seems that `xz` is preinstalled on the machine that runs the CI job. You copied this from the equivalent function for `pbzip2`. It works for `pbzip2` because that is not preinstalled and we have two jobs, one that runs normally and one where we install `pbzip2`.

I think we can "cheat" a bit by writing it this way, which will give full coverage:
```
    result = [], []
    if shutil.which("xz"):
        result = [PipedXzReader], [PipedXzWriter]
    return result
```

However, I still want to test that the fallback to `lzma.open` works if the `xz` binary is not available. For this, can you please add an `xz`-version of `test_falls_back_to_bzip2_open`? You’ll also need to add a `lacking_xz_permission` fixture.

(We should clean these fixtures up a bit, but no need to do it in this PR.)